autolens/pipeline/phase/abstract/phase.py

```diff
@@ -55,5 +55,5 @@ def make_result(self, result, analysis):
             use_as_hyper_dataset=self.use_as_hyper_dataset,
         )
 
-    def run(self, dataset, mask, results=None, positions=None):
+    def run(self, dataset, mask, results=af.ResultsCollection(), positions=None):
```

Comment:
Careful with this. This ResultsCollection will be used globally. It's better to default to None and then say self.results = results or af.ResultsCollection()